spec/controllers/users_controller_spec.rb

```diff
@@ -31,73 +31,117 @@
   end
 
   describe "POST create" do
-    before do
-      allow(Stripe::Charge).to receive(:create)
-    end
-
-    context "with valid credentials" do
+    context "with valid personal information" do
       after { ActionMailer::Base.deliveries.clear }
 
-      it "calls Stripe to charge user with subscription fees" do
-        expect(Stripe::Charge).to receive(:create)
-        post :create, user: Fabricate.attributes_for(:user)
+      before do
+        expect(StripeWrapper::Charge).to receive(:create).and_return(charge)
       end
 
-      it "creates new user" do
-        post :create, user: Fabricate.attributes_for(:user)
-        expect(User.count).to eq(1)
-      end
+      context "and valid credit card" do
+        let(:charge) do
+          double(:charge,
+                 successful?: true)
+        end
 
-      it "redirects to the sign-in page" do
-        post :create, user: Fabricate.attributes_for(:user)
-        expect(response).to redirect_to(sign_in_path)
+        it "creates new user" do
+          post :create, user: Fabricate.attributes_for(:user)
+          expect(User.count).to eq(1)
+        end
+
+        it "redirects to the sign-in page" do
+          post :create, user: Fabricate.attributes_for(:user)
+          expect(response).to redirect_to(sign_in_path)
+        end
+
+        context "with invitation" do
```

Comment:
This portion could be reduced with let/before. I suggest reviewing this whole PR and find places where you can reduce the size like that. It isn't always ideal for readability so the second part of this homework would be to decide which reductions are appropriate and which detract from the readability.